bot/exts/moderation/modlog.py

```diff
@@ -751,6 +759,64 @@ async def on_raw_message_edit(self, event: discord.RawMessageUpdateEvent) -> Non
             after_response, channel_id=Channels.message_log
         )
 
+    @Cog.listener()
+    async def on_thread_update(self, before: Thread, after: Thread) -> None:
+        """Log thread archiving, un-archiving and name edits."""
+        if before.name != after.name:
+            await self.send_log_message(
+                Icons.hash_blurple,
+                Colour.blurple(),
+                "Thread name edited",
+                (
+                    f"Thread {after.mention} (`{after.id}`) from {after.parent.mention} (`{after.parent.id}`): "
+                    f"`{before.name}` -> `{after.name}`"
+                )
+            )
+            return
+
+        if not before.archived and after.archived:
+            colour = Colour.red()
+            action = "archived"
+            icon = Icons.hash_red
+        elif before.archived and not after.archived:
+            colour = Colour.green()
+            action = "un-archived"
+            icon = Icons.hash_green
+        else:
+            return
+
+        await self.send_log_message(
+            icon,
+            colour,
+            f"Thread {action}",
+            f"Thread {after.mention} (`{after.id}`) from {after.parent.mention} (`{after.parent.id}`) {action}"
+        )
+
+    @Cog.listener()
+    async def on_thread_delete(self, thread: Thread) -> None:
+        """Log thread deletion."""
+        await self.send_log_message(
+            Icons.hash_red,
+            Colour.red(),
+            "Thread deleted",
+            f"Thread {thread.mention} (`{thread.id}`) from {thread.parent.mention} (`{thread.parent.id}`) deleted"
+        )
+
+    @Cog.listener()
+    async def on_thread_join(self, thread: Thread) -> None:
+        """Log thread creation."""
+        # If we are in the thread already we can most probably assume we already logged it?
+        # We don't really have a better way of doing this since the API doesn't make any difference between the two
+        if thread.me:
+            return
```

Comment:
![image](https://user-images.githubusercontent.com/38372706/133929721-230e5e74-732a-4de3-af0c-3c0975a6893f.png)

So this is sent when a thread is created, or we get access to a private thread by being mentioned in it.

We don't need this, when this event is received it should be from a thread being created no?

Reply:
I don't think you are looking at the right event. The API event for [`on_thread_join`](https://discordpy.readthedocs.io/en/master/api.html#discord.on_thread_join) seems to be [`Thread Create`](https://discord.com/developers/docs/topics/gateway#thread-create), which fires both when a new thread becomes accessible by the bot and when the bot actually joins it.

Without this check, we would log the thread twice, once when it is created and the even first fires, and a second time when it actually joins the thread. That's what I could also check during my testings.

Reply:
It also fites when _anyone_ joins a thread.

Reply:
Ahhh, I took a screenshot of the right event at first but then mixed it up.

Yes, Thread Update is the one that is sent when people join. But [Thread Create](https://discord.com/developers/docs/topics/gateway#thread-create) is only sent when a thread is made available to the bot.

![thread_create.jpg](https://user-images.githubusercontent.com/38372706/134139444-612a4f7c-2366-4ad0-add6-755631954cbb.jpg)

Reply:
yes, which I believe is the one used by `on_thread_join`, since its behavior matches that. So we do need to have this check.

Reply:
I don't understand, I am still not convinced. Thread Update is the one that gets sent when a thread gets updated, which it does when members join because now the list of joined members change.

Reply:
Indeed, but I am not sure how that's relevant? I could confirm through testing that's the behaviour we want